multihost_test/Bugzillas/test_Regression.py

```diff
@@ -0,0 +1,106 @@
+""" Automation of Shadow Utils tests """
+
+from __future__ import print_function
+import pytest
+import subprocess
+import os
+import time
+from sssd.testlib.common.utils import SSHClient, sssdTools
+
+
+def execute_cmd(multihost, command):
+    """ Run command on client machine """
+    cmd = multihost.client[0].run_command(command)
+    return cmd
+
+
+class TestShadowUtilsRegression():
+    """
+    Automation of Shadow Utils tests
+    """
+    @pytest.mark.tier1
+    def test_bz507706(self, multihost, create_backup):
+        """
+        :title:bz507706 grpconv wants to take all memory if /etc/group contains duplicate entries
+        :id: dfe8883e-6728-11ee-b0c7-845cf3eff344
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=507706
+        :steps:
+          1. Add 10 user groups named group_0 to group_9.
+          2. Checks the number of groups with the name pattern group_ in the /etc/group file.
+            It asserts that there should be 10 such groups.
+          3. The content of the /etc/group file is saved into /tmp/anuj.
+          4. The groups with the name pattern group_ from /tmp/anuj are appended back into the /etc/group file.
+            This effectively doubles the number of such groups.
+          5. Checks again and asserts that the number of groups with the name pattern group_ in
+            the /etc/group file should now be 20.
+          6. The grpconv command is executed in the background (indicated by &).
+            This command initializes the shadow file.
+          7. Run the grpck command, which checks the integrity of password and group files.
+            If there's a mismatch, it prompts for a confirmation to fix it.
+            Here, the script provides 10 "yes" responses to fix potential mismatches.
+          8. Delete the 10 user groups it had previously created.
+        :expectedresults:
+          1. Should succeed
+          2. Should succeed
+          3. Should succeed
+          4. Should succeed
+          5. Should succeed
+          6. Should succeed
+          7. Should succeed
+          8. Should succeed
+        """
+        for i in range(10):
+            execute_cmd(multihost, f"groupadd group_{i}")
+        assert execute_cmd(multihost, "grep group_ /etc/group | wc -l").stdout_text.split('\n')[0] == '10'
+        execute_cmd(multihost, "cat /etc/group > /tmp/anuj")
+        execute_cmd(multihost, "grep group_ /tmp/anuj >> /etc/group")
+        assert execute_cmd(multihost, "grep group_ /etc/group | wc -l").stdout_text.split('\n')[0] == '20'
+        execute_cmd(multihost, "grpconv &")
+        execute_cmd(multihost, "sleep 20")
```

Comment:
In order to call `grpconv` in the foreground you need to remove the `&`. Once that is done the call should be done synchronously, which means that it won't return until it has finished. If that's the case, then you can also remove the `sleep`.

Reply:
Fixed now